Manito/Manito/Screens/Detail-Wait/UIComponent/DetailHalfModalController.swift

```diff
@@ -233,5 +275,93 @@ class DetailModalController: BaseViewController {
+                makeAlert(title: "인원 수 제한", message: "최대 7일까지 선택가능해요 !")
+            } else {
+                setDateRange()
+                calendar.reloadData()
+            }
+        } else if calendar.selectedDates.count > 2 {
+            for _ in 0 ..< calendar.selectedDates.count - 1 {
+                calendar.deselect(calendar.selectedDates[0])
+            }
+            selectStartDate = date
+            calendar.select(selectStartDate)
+            calendar.reloadData()
+        }
+    }
+
+    func calendar(_ calendar: FSCalendar, didDeselect date: Date, at monthPosition: FSCalendarMonthPosition) {
+        for _ in 0 ..< calendar.selectedDates.count {
+            calendar.deselect(calendar.selectedDates[0])
+        }
+        selectStartDate = date
+        calendar.select(date)
+        calendar.reloadData()
+    }
+
+    func calendar(_ calendar: FSCalendar, shouldSelect date: Date, at monthPosition: FSCalendarMonthPosition) -> Bool {
+        if date < Date() - 86400 {
+            makeAlert(title: "과거로 가시게요..?", message: "오늘보다 이전 날짜는 \n 선택하실 수 없어요 !")
+            return false
+        } else {
+            return true
+        }
+    }
+
+    func calendar(_ calendar: FSCalendar, appearance: FSCalendarAppearance, titleDefaultColorFor date: Date) -> UIColor? {
+        if date < Date() - 86400 {
+            return .grey005.withAlphaComponent(0.4)
+        } else if isFirstTap == false {
+            return .white
+        } else if date < selectStartDate + 604800 && date > selectStartDate - 604800 {
+            return .white
+        } else if calendar.selectedDates.count > 2 {
+            return .white
+        } else {
+            return .grey005.withAlphaComponent(0.6)
+        }
+    }
+
+    // MARK: - func
+
+    func setDateRange() {
+        if countDateRange() <= 7 {
+            var startDate: Date
+            if calendar.selectedDates[0] < calendar.selectedDates[1] {
+                startDate = calendar.selectedDates[0]
+                while startDate < calendar.selectedDates[1] {
+                    startDate = Calendar.current.date(byAdding: .day, value: 1, to: startDate)!
+                    calendar.select(startDate)
+                }
+            } else {
+                startDate = calendar.selectedDates[1]
+                while startDate < calendar.selectedDates[0] {
+                    startDate = Calendar.current.date(byAdding: .day, value: 1, to: startDate)!
+                    calendar.select(startDate)
+                }
+            }
+        }
+    }
+
+    func countDateRange() -> Int {
+        if calendar.selectedDates[0] < calendar.selectedDates[1] {
+            let rangeCount = calendar.selectedDates[1].timeIntervalSince(calendar.selectedDates[0]) / 86400
+            return Int(rangeCount) + 1
+        } else {
+            let rangeCount = calendar.selectedDates[0].timeIntervalSince(calendar.selectedDates[1]) / 86400
+            return Int(rangeCount) + 1
+        }
```

Comment:
p5;

```suggestion
       let clickedPastDate = calendar.selectedDates[0] < calendar.selectedDates[1]
       let selectedDate = clickedPastDate ?  calendar.selectedDates[1].timeIntervalSince(calendar.selectedDates[0]) :   calendar.selectedDates[0].timeIntervalSince(calendar.selectedDates[1]) 
       let rangeCount = selectedDate / 86400
       
       return Int(rangeCount) + 1
```
이렇게 쓰면 읽기도 좋고 중복도 줄어들 거 같아여!
변수명은 대충 제가 이런 의미의 변수인 거 같다 싶은걸로 지어서.. 무시해주시죠👀

Reply:
이렇게 또 한 수 배워갑니다 ...